### PR TITLE
DRY shaderc_test.cc

### PR DESCRIPTION
**Pull request by Qining**
Add new subclass CompileString : public testing::Test to wrap initialization of compiler, options etc.

### CODE REVIEW

File: libshaderc/src/shaderc_test.cc [hunk 1]
```diff
@@ -111,327 +114,299 @@ class Compiler {
   shaderc_compiler_t compiler;
 };
 
-// Compiles a shader and returns true on success, false on failure.
-bool CompilationSuccess(const shaderc_compiler_t compiler,
-                        const std::string& shader, shaderc_shader_kind kind) {
-  return shaderc_module_get_success(
-      Compilation(compiler, shader, kind).result());
-}
+// A testing class to test the compilation of string with or without options.
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: Good blurb.
Minor grammar problem:  should be  "test the compilation of a string".  (Was missing "a")

File: libshaderc/src/shaderc_test.cc [hunk 2]
```diff
@@ -535,26 +495,22 @@ TEST(CompileKinds, Geometry) {
...
-                                 shaderc_glsl_tess_control_shader));
+  EXPECT_TRUE(CompilationSuccess(kTCSShader, shaderc_glsl_tess_control_shader));
 }
 
-TEST(CompileKinds, TessEvaluation) {
-  Compiler compiler;
-  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+TEST_F(CompileKindsTest, TessEvaluation) {
+  ASSERT_NE(nullptr, compiler_.get_compiler_handle());
   const std::string kTESShader =
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: Similarly, should be kTEShader?

File: libshaderc/src/shaderc_test.cc [hunk 3]
```diff
@@ -535,26 +495,22 @@ TEST(CompileKinds, Geometry) {
...
-                                 shaderc_glsl_geometry_shader));
+  EXPECT_TRUE(CompilationSuccess(kGeoShader, shaderc_glsl_geometry_shader));
 }
 
-TEST(CompileKinds, TessControl) {
-  Compiler compiler;
-  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+TEST_F(CompileKindsTest, TessControl) {
+  ASSERT_NE(nullptr, compiler_.get_compiler_handle());
   const std::string kTCSShader =
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: Should be kTCShader, for Tesselleation Control Shader?   What's the second S in kTCSShader?

Better to just write it out:  kTessControlShader

File: libshaderc/src/shaderc_test.cc [hunk 4]
```diff
@@ -111,327 +114,299 @@ class Compiler {
...
+      const std::string& shader, shaderc_shader_kind kind,
+      const shaderc_compile_options_t options = nullptr) {
+    const Compilation comp(compiler_.get_compiler_handle(), shader, kind,
+                           options);
+    EXPECT_TRUE(shaderc_module_get_success(comp.result())) << kind << '\n'
+                                                           << shader;
+    return shaderc_module_get_error_message(comp.result());
+  };
+
+  // Compiles a shader, expects compilation fail, and returns the warning
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: "exepcts compilation failure"

**Comment by dneto0** [comment]
-1.  Minor wording issues.

**Comment by dneto0** [comment]
+2

**Comment by Qining** [comment]
Merged into master commit: 4fd51cc

**Pull request closed without merging**